python-avd/pyavd/_eos_designs/structured_config/overlay/router_bfd.py

```diff
@@ -17,10 +19,9 @@ class RouterBfdMixin(Protocol):
     Class should only be used as Mixin to a AvdStructuredConfig class.
     """
 
-    @cached_property
-    def router_bfd(self: AvdStructuredConfigOverlayProtocol) -> dict | None:
-        """Return structured config for router_bfd."""
+    @structured_config_contributor
+    def router_bfd(self: AvdStructuredConfigOverlayProtocol) -> None:
+        """Set structured config for router_bfd."""
         if self.shared_utils.overlay_cvx:
-            return None
-
-        return {"multihop": self.inputs.bfd_multihop._as_dict()}
+            return
+        self.structured_config.router_bfd.multihop = self.inputs.bfd_multihop._cast_as(EosCliConfigGen.RouterBfd.Multihop)
```

Comment:
shaving off some runtime:
```suggestion
        if self.inputs.bfd_multihop:
            self.structured_config.router_bfd.multihop = self.inputs.bfd_multihop._cast_as(EosCliConfigGen.RouterBfd.Multihop)
```